colossus/src/main/scala/colossus/service/ServiceServer.scala

```diff
@@ -89,11 +88,14 @@ extends Controller[I,O](codec, ControllerConfig(50)) {
   private val requestBuffer = collection.mutable.Queue[SyncPromise]()
   private var numRequests = 0
 
-  def idleCheck(period: Duration) {
+  override def idleCheck(period: Duration) {
+    super.idleCheck(period)
```

Comment:
Yeah I realize that "hoping" the parent method gets invoked leaves some room for error, but I think starting to put hooks everywhere might end up being just as error prone.  At least we're only doing it in a few places and it's something that's easily testable.